src/utils.jl

```diff
@@ -8,7 +8,7 @@ _indexlength(i::Integer) = i
 _indexlength(i::Colon) = Colon()
 
 _offset(axparent::AbstractUnitRange, ax::AbstractUnitRange) = first(ax) - first(axparent)
-_offset(axparent::AbstractUnitRange, ax::Integer) = 1 - first(axparent)
+_offset(axparent::AbstractUnitRange, ::Union{Integer, Colon}) = 1 - first(axparent)
```

Comment:
We can let `OffsetArray` constructor to decide how to deal with `Colon`:

```suggestion
_offset(axparent::AbstractUnitRange, ::Integer) = 1 - first(axparent)
_offset(axparent::AbstractUnitRange, ::Colon) = Colon()
```

Reply:
We may do this, but this would mean interpreting `Colon` as an offset as opposed to an axis as we do now. So after this a `Tuple{Vararg{Union{Integer, Colon}}}` may be seen as offsets and other combinations (eg. `Colon`s and `AbstractUnitRange`s, `CartesianIndices` etc) as axes. I think it should be clear from the context what the meaning of the colon is.